Possible variable re-use

mujuonly opened the pull request:
The variable `$spam` is set above in the function like `$spam = empty( $userdata['spam'] ) ? 0 : (bool) $userdata['spam'];` We can re-use the variable instead of checking again as the specific key in the $userdata array is not getting changed.

<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->

Trac ticket: https://core.trac.wordpress.org/ticket/58152

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

mujuonly committed:
Re-use variable

The variable `$spam` is set above in the function like `$spam = empty( $userdata['spam'] ) ? 0 : (bool) $userdata['spam'];`
We can re-use the variable instead of checking again as the specific key in the $userdata array is not getting changed.

todeveni commented on src/wp-includes/user.php:
```diff
@@ -2460,8 +2460,8 @@ function wp_insert_user( $userdata ) {
-		if ( isset( $userdata['spam'] ) && $userdata['spam'] != $old_user_data->spam ) {
-			if ( 1 == $userdata['spam'] ) {
+		if ( $spam !== $old_user_data->spam ) {
+			if ( 1 === $spam ) {
```
`$spam` is falsy `0` or a boolean value, so this should be `if ( true === $spam ) {`

mujuonly committed:
Updates the value to the boolean

`$spam` is falsy 0 or a boolean value, so this should be if ( true === $spam ) {

costdev reviewed (approved):
Thanks for the PR @mujuonly and for the update as suggested by @todeveni. This looks good to me 👍
